presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1570,6 +1571,7 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             // ORDER BY should "see" both output and FROM fields during initial analysis and non-aggregation query planning
             Scope orderByScope = Scope.builder()
                     .withParent(sourceScope)
+                    .withScopeId(outputScope.getScopeId())
```

Comment:
Dear Reviewers,

this is weak. A new scope is created but with old id, so that `ResolvedFields` produced by it will be indistinguishable from those produced by `outputScope`. They key here is the comment above

> ORDER BY should "see" both output and FROM fields ...

But `Scope` doesn't not support multiple parents. Oh, and this even is _not_ multiple parents, since we effectively consider `outputScope`, but not `outputScope.parent` in resolution.

I _think_ some extenstion to `Scope` is needed. What do you think?